### PR TITLE
add range support to std.string.stripLeft

### PR DESCRIPTION
**Pull request by WalterBright**
unit test coverage 100%

### CODE REVIEW

**Comment by JakobOvrum** [comment]
std.algorithm.mutation already has generic `strip`, `stripLeft` and `stripRight`. The string versions that use `isWhite` should probably be composed in terms of the generic ones?

**Comment by WalterBright** [comment]
Unfortunately, the generic strip does not decode ranges of char or wchar:
```
import std.stdio;
import std.algorithm;
import std.utf;
import std.array;

void main() {
    assert("ëëêéüŗōpéêëë".strip('ë') == "êéüŗōpéê"); // works
    assert("ëëêéüŗōpéêëë".byCodeUnit.strip('ë').array == "êéüŗōpéê"); // fails
}
```

File: std/string.d [hunk 1]
```diff
@@ -2491,6 +2509,11 @@ C[] stripLeft(C)(C[] str) @safe pure @nogc
            "hello world" ~ [lineSep]);
     assert(stripLeft([paraSep] ~ "hello world" ~ paraSep) ==
            "hello world" ~ [paraSep]);
+
+    import std.utf : byChar;
+    import std.array;
+    assert(stripLeft("     hello world     "w.byChar).array ==
```
Comment (JesseKPhillips) [inline std/string.d]: Instead of calling .array, you should be able to use .equal to verify equality and you won't have an allocation.

Comment (WalterBright) [inline std/string.d]: Being a test case, the allocation doesn't matter.

**Comment by JakobOvrum** [comment]
While I would prefer changing std.algorithm's strip functions and deferring to those, the interface would be the same and this is a tiny patch, so LGTM.

**Comment by JakobOvrum** [comment]
Auto-merge toggled on

**Comment by aG0aep6G** [comment]
This introduced a regression: https://issues.dlang.org/show_bug.cgi?id=14827

**Pull request merged**